Presscotts_Mortar_Calculator_(no_envirement)_V0.2.py

```diff
@@ -19,215 +18,225 @@
    air density 1pct incline]
 '''
 
-import math 
+import math
+
+
 def getList(dict):
     list = []
     for key in dict.keys():
         list.append(key)
-         
+
     return list
+
+
 def checkKey(dic, key):
     if key in dic.keys():
-        print("Present, ", end =" ")
+        print("Present, ", end=" ")
         print("value =", dic[key])
     else:
         print("Not present")
-        
+
 
 def main():
     '''mortar range table for reference for math'''
     charge0 = {

```

Comment:
Future: It would be worth creating a user-defined type to clarify these tables.
Suggestion:

```
class RangeTableEntry
{
range:            int,
elevation:        int,
time_of_flight:   int,
...etc.
}
class RangeTable
    0...n RangeTableEntry
```

This way we can use more meaningful lookup modes by defining class methods. i.e. Determining elevation could look something like: 
`firingSolution = charge1Table.getSolution(range)`
then
`firingSolution.getTimeOfFlight()`
`firingSolution.getElevation()`

Reply:
The idea in how this would work is very similar to how the dictionaries work

It would make the code more readable to outside user but ultimately i dont think its necessary 

Would also require a complete redo of a large chunk of computation

The index of the keys is defined in the above 

C1.getTimeOfFlight in the current form is
Charge1[range][3]

I do agree it would be nice but i don't  think it really adds much in terms of functionality, it only really helps readability however redoing the calculations to a functions and calling them would definitely clean up the code by a large degree

Reply:
Certainly similar - yeah. It will also likely add a small bit of performance overhead and some size to the file.
This change would only address maintainability for future use and look good on a CV some day as a way of saying "look how clean my code is - everyone would love to have my code in their team's toolkit" :).